### ISSUE TITLE
Mono/Csharp: Core\Basis.cs { or ; expected syntax errors (Can't Build Projects)

### ISSUE THREAD

**Issue by exts** [godotengine/godot#17936]
**Godot version:**
Master https://github.com/godotengine/godot/commit/59710880cd21d79ae1329f04eb35ea48e646acb3 

**OS/device including version:**
Windows 10 (64 bit)

**Issue description:**
Getting syntax error from the builds menu

**Steps to reproduce:**
1. Create any csharp project
2. Attempt to build
3. Get errors

```
Core\Basis.cs(52, 17): { or ; expected
Core\Basis.cs(52, 17): } expected
Core\Basis.cs(54, 10): } expected
Core\Basis.cs(58, 17): { or ; expected
Core\Basis.cs(58, 17): { or ; expected
Core\Basis.cs(60, 10): { or ; expected
Core\Basis.cs(64, 17): { or ; expected
Core\Basis.cs(64, 17): { or ; expected
Core\Basis.cs(66, 10): { or ; expected
```

**Other:**

This commit could be the source of the issue: https://github.com/godotengine/godot/commit/91f271fa9e7f1dd0fc36f434c4b7a795a8c463c0#diff-e06876bd0e6ae64597e7e647454eb74b

Also built using visual studio 2017 build tools, scons 3.0.0, python 2.7, mono 5.8 (shouldn't matter, never had this issue before)

**Comment by nazgulsenpai** [comment]
Same issue here on most recent master branch, Windows 10 64bit. Engine compiles properly with Mono, but upon launching editor and building GodotSharp Solution, Basis.cs errors.

**Comment by exts** [comment]
Alright I was testing some builds and found this bug to build normally: https://github.com/godotengine/godot/commit/bbd21c22030c85d098fa79ee69f78803a86c15ba

So yeah I believe somewhere after march 23rd the builds stopped working.

**Comment by mysticfall** [comment]
I can't reproduce this problem, but probably it'd be worthwhile to check what .NET version `GodotSharp.csproj` is referencing for those who have the problem.

In my case, `GodotSharp.csproj` contains the following:
```xml
<TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
```

And I tested it with the latest master today, with Mono `5.13.0.132`.

**Comment by neikeq** [comment]
What's the MSBuild version you have installed? IINW, expression bodied properties are a C#7 thing (requires MSBuild 15.0). We should keep the API assemblies to C#6 for now though...

**Comment by mysticfall** [comment]
In my case, it's MSBuild `15.4` so probably that explains the differernce then.

**Comment by nazgulsenpai** [comment]
I built with VS2017 (sorry I don't have the MSbuild ver in front of me) and have had no issue running from source builds until today, when I experienced this issue. Did something change between 3/21 (when I built the last working version) to today that would cause this issue?

**Comment by Alexander-Alekseev** [comment]
Same problem here.
Mono: 5.4.1.7 and 5.10.0.160.
MSBuild: 14.0.25420.1.
TargetFrameworkVersion: v4.5.

Temp fix: in GodotSharp\Core\Basis.cs replace this -
```
        public Vector3 x
        {
            get => GetAxis(0);
            set => SetAxis(0, value);
        }

        public Vector3 y
        {
            get => GetAxis(1);
            set => SetAxis(1, value);
        }

        public Vector3 z
        {
            get => GetAxis(2);
            set => SetAxis(2, value);
        }
```
replaced with -
```
	public Vector3 x
        {
          get { return GetAxis(0); }
          set { SetAxis(0, value); }
        }

        public Vector3 y
        {
          get { return GetAxis(1); }
          set { SetAxis(1, value); }
        }

        public Vector3 z
        {
          get { return GetAxis(2); }
          set { SetAxis(2, value); }
        }
```

**Comment by Alexander-Alekseev** [comment]
Because of 91f271fa9e7f1dd0fc36f434c4b7a795a8c463c0 (2018-03-08), right?

https://github.com/godotengine/godot/blob/91f271fa9e7f1dd0fc36f434c4b7a795a8c463c0/modules/mono/glue/cs_files/Basis.cs#L50-L71

**Comment by cart** [comment]
Yeah this was my fault and everything said above is correct.  Compatibility should definitely be favored over syntactic sugars.

**Comment by exts** [comment]
Going to clear up some stuff.

When I build godot the editor uses "MSBuild(System)" on **windows**, MSBuild (Mono) errors out.

- Mono version: 5.10.x Stable
- MSbuild Version: 15, thing is Idk where godot is pulling from. I have VS Build Tools 2017 & dotNet 2 installed which both uses a different version of MSBuild 15.

**Comment by akien-mga** [comment]
Fixed by #17940.
